util/util.go

```diff
@@ -291,22 +290,44 @@ func MachineToInfrastructureMapFunc(gvk schema.GroupVersionKind) handler.ToReque
 
 // HasOwnerRef returns true if the OwnerReference is already in the slice.
 func HasOwnerRef(ownerReferences []metav1.OwnerReference, ref metav1.OwnerReference) bool {
-	for _, r := range ownerReferences {
-		if reflect.DeepEqual(ref, r) {
-			return true
-		}
-	}
-	return false
+	return FindOwnerRef(ownerReferences, ref) > -1
 }
 
 // EnsureOwnerRef makes sure the slice contains the OwnerReference.
 func EnsureOwnerRef(ownerReferences []metav1.OwnerReference, ref metav1.OwnerReference) []metav1.OwnerReference {
-	if !HasOwnerRef(ownerReferences, ref) {
+	idx := FindOwnerRef(ownerReferences, ref)
+	if idx == -1 {
 		return append(ownerReferences, ref)
 	}
+	ownerReferences[idx] = ref
 	return ownerReferences
 }
 
+// FindOwnerRef returns the index of the owner reference in the slice if found, or -1.
+func FindOwnerRef(ownerReferences []metav1.OwnerReference, ref metav1.OwnerReference) int {
```

Comment:
Might be good to rename this to make it clear it is returning an index rather than an OwnerRef.

Reply:
Also, any reason to not make this private?

Reply:
yeah sgtm